lib/reverse_sentence.rb

```diff
@@ -1,6 +1,30 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+  
+  len = my_sentence.length
+  
+  (len / 2).times do |i|
+    temp = my_sentence[len - i - 1]
+    my_sentence[len - i - 1] = my_sentence[i]
+    my_sentence[i] = temp
```

Comment:
Notice that you're reversing substrings multiple times.  That makes it a great candidate for a helper method.